tests/system_tests/test_blueapi_system.py

```diff
@@ -0,0 +1,260 @@
+
+    with pytest.raises(BlueskyRemoteControlError) as exception:
+        client.pause()
+        assert str(exception) == "<Response [400]>"
+
+
+def test_get_task_by_status(client: BlueapiClient):
+    task_1 = client.create_task(_SIMPLE_TASK)
+    task_2 = client.create_task(_SIMPLE_TASK)
+    task_by_pending_request = requests.get(
+        client._rest._url("/tasks"), params={"task_status": TaskStatusEnum.PENDING}
+    )
+    assert task_by_pending_request.status_code == status.HTTP_200_OK
+    task_by_pending = TypeAdapter(TasksListResponse).validate_python(
+        task_by_pending_request.json()
+    )
+
+    assert len(task_by_pending.tasks) == 2
+    for task in task_by_pending.tasks:
+        trackable_task = TypeAdapter(TrackableTask).validate_python(task)
+        assert trackable_task.is_complete is False and trackable_task.is_pending is True
+
+    client.start_task(WorkerTask(task_id=task_1.task_id))
+    while not client.get_task(task_1.task_id).is_complete:
+        time.sleep(0.1)
+    client.start_task(WorkerTask(task_id=task_2.task_id))
+    while not client.get_task(task_2.task_id).is_complete:
+        time.sleep(0.1)
+    task_by_completed_request = requests.get(
+        client._rest._url("/tasks"), params={"task_status": TaskStatusEnum.COMPLETE}
+    )
+    task_by_completed = TypeAdapter(TasksListResponse).validate_python(
+        task_by_completed_request.json()
+    )
+    assert len(task_by_completed.tasks) == 2
+    for task in task_by_completed.tasks:
+        trackable_task = TypeAdapter(TrackableTask).validate_python(task)
+        assert trackable_task.is_complete is True and trackable_task.is_pending is False
+
+    client.clear_task(task_id=task_1.task_id)
+    client.clear_task(task_id=task_2.task_id)
+
+
+def test_progress_with_stomp(client_with_stomp: BlueapiClient):
+    all_events: list[AnyEvent] = []
+
+    def on_event(event: AnyEvent):
+        all_events.append(event)
+
+    client_with_stomp.run_task(_SIMPLE_TASK, on_event=on_event)
+    assert isinstance(all_events[0], WorkerEvent) and all_events[0].task_status
+    task_id = all_events[0].task_status.task_id
+    running_event = WorkerEvent(
+        state=WorkerState.RUNNING,
+        task_status=TaskStatus(
+            task_id=task_id,
+            task_complete=False,
+            task_failed=False,
+        ),
+    )
+    pending_event = WorkerEvent(
+        state=WorkerState.IDLE,
+        task_status=TaskStatus(
+            task_id=task_id,
+            task_complete=False,
+            task_failed=False,
+        ),
+    )
+    complete_event = WorkerEvent(
+        state=WorkerState.IDLE,
+        task_status=TaskStatus(
+            task_id=task_id,
+            task_complete=True,
+            task_failed=False,
+        ),
+    )
+    assert running_event in all_events
+    assert pending_event in all_events
+    assert complete_event in all_events
+    assert len(all_events) == 3
```

Comment:
```suggestion
    assert all_events == [
    WorkerEvent(
        state=WorkerState.RUNNING,
        task_status=TaskStatus(
            task_id=task_id,
            task_complete=False,
            task_failed=False,
        ),
    ),
    WorkerEvent(
        state=WorkerState.IDLE,
        task_status=TaskStatus(
            task_id=task_id,
            task_complete=False,
            task_failed=False,
        ),
    ),
    WorkerEvent(
        state=WorkerState.IDLE,
        task_status=TaskStatus(
            task_id=task_id,
            task_complete=True,
            task_failed=False,
        ),
    )]
```

I'd go further and explicitly check that they arrived in exactly this order